nautobot/virtualization/filters.py

```diff
@@ -182,18 +215,48 @@ class VirtualMachineFilterSet(NautobotFilterSet, LocalContextFilterSet, TenancyF
         method="_has_primary_ip",
         label="Has a primary IP",
     )
-    tag = TagFilter()
+    primary_ip4 = MultiValueCharFilter(
+        method="filter_primary_ip4",
+        label="Primary IP Address",
+    )
+    primary_ip6 = MultiValueCharFilter(
+        method="filter_primary_ip6",
+        label="Primary IP Address",
+    )
+    services = django_filters.ModelMultipleChoiceFilter(
+        field_name="services", queryset=Service.objects.all(), label="Services (ID)"
+    )
+    has_services = RelatedMembershipBooleanFilter(
+        field_name="services",
+        label="Has services",
+    )
+    vm_interfaces = django_filters.ModelMultipleChoiceFilter(
+        field_name="interfaces", queryset=VMInterface.objects.all(), label="VMInterfaces (ID)"
+    )
+    has_vm_interfaces = RelatedMembershipBooleanFilter(
+        field_name="interfaces",
+        label="Has VMinterfaces",
+    )
```

Comment:
Thinking ahead toward #1178, I wonder if it would be better for these to be `interfaces` and `has_interfaces` instead?